lib/util.js

```diff
@@ -51,6 +53,37 @@ const validConditionList = {
     'latency': ['>', '<'],
     'error': ['>']
 };
+function httpClientRetries(urlSuffix, header, method, retries = 1, content) {
+    return __awaiter(this, void 0, void 0, function* () {
+        let httpResponse;
+        try {
+            if (method == 'get') {
+                httpResponse = yield httpClient.get(urlSuffix, header);
+            }
+            else if (method == 'del') {
+                httpResponse = yield httpClient.del(urlSuffix, header);
+            }
+            else {
+                httpResponse = yield httpClient.request(method, urlSuffix, content, header);
+            }
+            if (httpResponse.message.statusCode != undefined && [408, 404, 429, 502, 503, 504].includes(httpResponse.message.statusCode)) {
+                throw { message: httpResponse.message.statusMessage }; // throwing as message to catch it as err.message
+            }
+            return httpResponse;
+        }
+        catch (err) {
+            if (retries) {
+                let sleeptime = (5 - retries) * 1000 + Math.floor(Math.random() * 5001);
```

Comment:
how did we arrive at this function for sleep?

Reply:
1st retries = 3, cuz i used recursion, when the number of retries remining getting decreased then we are increasing the await time